src/Channel.php

```diff
@@ -1149,4 +1190,16 @@ static function (Protocol\AbstractFrame $frame) use ($deferred) {
 
         return $deferred->promise();
     }
+
+    /**
+     * Channel was closed by server
+     */
+    public function forceClose(\Throwable $exception): void
+    {
+        if ($this->state !== self::STATE_CLOSED) {
+            $this->state = self::STATE_CLOSED;
+            $this->commandWaitQueue->cancel($exception);
```

Comment:
This step is key to notify all waiting commands if sth happens with the connection / client / channel. It avoid stale promises